github/resource_github_organization_ruleset.go

```diff
@@ -588,57 +589,70 @@ func resourceGithubOrganizationRuleset() *schema.Resource {
 func resourceGithubOrganizationRulesetCreate(d *schema.ResourceData, meta any) error {
 	client := meta.(*Owner).v3client
 
-	owner := meta.(*Owner).name
-
-	rulesetReq := resourceGithubRulesetObject(d, owner)
+	rulesetReq := resourceGithubRulesetObject(d, meta.(*Owner).name)
 
+	org := meta.(*Owner).name
 	ctx := context.Background()
 
-	var ruleset *github.Ruleset
+	var ruleset *github.RepositoryRuleset
 	var err error
 
-	ruleset, _, err = client.Organizations.CreateOrganizationRuleset(ctx, owner, rulesetReq)
+	ruleset, _, err = client.Organizations.CreateRepositoryRuleset(ctx, org, *rulesetReq)
 	if err != nil {
 		return err
 	}
 	d.SetId(strconv.FormatInt(*ruleset.ID, 10))
+
 	return resourceGithubOrganizationRulesetRead(d, meta)
 }
 
 func resourceGithubOrganizationRulesetRead(d *schema.ResourceData, meta any) error {
 	client := meta.(*Owner).v3client
 
-	owner := meta.(*Owner).name
-
+	org := meta.(*Owner).name
 	rulesetID, err := strconv.ParseInt(d.Id(), 10, 64)
 	if err != nil {
 		return unconvertibleIdErr(d.Id(), err)
 	}
 
-	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+	ctx := context.WithValue(context.Background(), ctxId, rulesetID)
 	if !d.IsNewResource() {
 		ctx = context.WithValue(ctx, ctxEtag, d.Get("etag").(string))
 	}
 
-	var ruleset *github.Ruleset
+	var ruleset *github.RepositoryRuleset
 	var resp *github.Response
 
-	ruleset, resp, err = client.Organizations.GetOrganizationRuleset(ctx, owner, rulesetID)
+	ruleset, resp, err = client.Organizations.GetRepositoryRuleset(ctx, org, rulesetID)
 	if err != nil {
 		ghErr := &github.ErrorResponse{}
```

Comment:
This code pattern looks to be fundamentally flawed as we've got an error and are not handling it, I suspect we'd find it throughout the codebase. In this case we don't return the error and the ruleset is empty so we remove it from state which can result in issues like #2889.